### DISCUSSION TITLE
Missing(?) Options for Bulk editing Stock

### DISCUSSION THREAD

**Discussion by wieselukas** [inventree/InvenTree#8394] [General]
Hi everyone,

I am looking for a way to bulk edit stock quantities in Inventree. I have some ideas, but none really got me what I wanted. 
The background is that we often send all StockItems of a Part to the assembly house and receive back what was not consumed. During these times the stock level should be properly reflected, so a list of parts needs to be removed from stock (or location changed) and then another list needs to be added again. Doing this for ~100 parts each on the specific parts page is quite tedious, so not the ideal option.

What made me write here, is that inventree has very nice ways of bulk editing stock. By selecting multiple items from the Stock Item List, one get's a nice window where stock of all items can be edited in bulk. But selecting the parts seems to be the issue. So the functionality I'd like doesn't seem to be too far away (;

Some Ideas I had:
- **using Build orders:** Build orders nicely implement removing a BOM. But this won't allow adding stock and also semantically it doesn't quite match the idea of sending stock off to an assembly house, when you expect to get most of it back.
- **using regex search:** Enabling regex search, one can filter for a list of words with (word1|word2). So my idea was to put all my Part IPNs in such a list, filter by them and thus get the option of selecting all and bulk editing all. But regex search only works on the Status bar search and not in the Stock Item search. And the Status bar search results won't allow me to bulk edit stock.

Am I missing something? If not, maybe this could be the starting point for implementing either regex search in the Stock Item List or add the bulk editing functionality in the Search results page. Or maybe an "upload IPN filter list" button could be an option too. 

Lukas

**Comment by SchrodingersGat** [comment]
@wieselukas to clarify, you are are looking for improvements for filtering / selecting the items in the "Stock Item Table", to allow you finer control over which ones you are editing?

**Reply by wieselukas** [reply]
That would be one way of getting done what I need, yes! But the question is also if I overlooked some feature or if anybody has some creative idea of how to get this done. Writing a simple command line tool using the api would also be an option I guess.

**Comment by spamik** [comment]
I'm looking for something similar. I want to use Inventree as inventory management for company internal material - something like ethernet patch cables and so on. For this I don't need tracking of orders and so, I just need to know how many of things are on stock (so I know when I need to order new). I have a barcode scanner and label printer, so idea is that every stock part will have printed label and anybody who is using it just scan the items which he is taking out. At the moment is needed to scan every particular item, for every item select "remove stock", type number and confirm it. And do the same for next thing and so on...

It would be nice just have something like shopping cart, scan with barcode scanner all things which person is taking out, just write note and after that it will do remove stock operation on all those things. (Add stock operation can work in the same manner as @wieselukas is asking :-) ).
